frontend/src/routes/Login.tsx

```diff
@@ -50,37 +50,46 @@ const Login: React.FC<Props> = ({ queryRef }) => {
     useNoindexTag();
     const { t } = useTranslation();
     const { currentUser } = usePreloadedQuery(query, queryRef);
+    const router = useRouter();
+    const isLoggedIn = currentUser !== null;
 
-    return <Outer>
-        <div css={{
-            height: "calc(1.7 * var(--header-height))",
-            maxHeight: "35vh",
-            padding: `${HEADER_BASE_PADDING}px 0`,
-            textAlign: "center",
-        }}><Logo /></div>
+    React.useEffect(() => {
+        if (isLoggedIn) {
+            // This isn't working - 
+            // window.sessionStorage.getItem(REDIRECT_STORAGE_KEY) is always null.
+            const redirectTo = window.sessionStorage.getItem(REDIRECT_STORAGE_KEY) ?? "/";
+            window.sessionStorage.removeItem(REDIRECT_STORAGE_KEY);
+            router.goto(redirectTo, true);
+        }
+    });
 
-        <main css={{
-            margin: "0 auto",
-            padding: 16,
-            maxWidth: "100%",
-            flexGrow: 1,
-            display: "flex",
-            flexDirection: "column",
-        }}>
-            {currentUser !== null
-                ? <AlreadyLoggedIn displayName={currentUser.displayName} />
-                : <>
-                    <PageTitle title={t("user.login")} />
-                    <LoginBox />
-                    <div css={{ marginTop: 12, fontSize: 14, lineHeight: 1 }}>
-                        <BackButton />
-                    </div>
-                </>
-            }
-        </main>
+    return isLoggedIn
+        // Don't render anything when a redirect is triggered.
+        ? null
+        : <Outer>
```

Comment:
From testing, it seems that a logged in user never sees the login page, so I would just remove this extra case. Just always render the login form. If we notice that in practice, the login page briefly flashes, we can still add this. But I don't think it's necessary.

Reply:
For me, it does indeed flash very briefly, that's why I added this.

Reply:
Oh hu. I guess then lets leave it. Tho of course then we briefly flash "nothingness". But I guess thats better than a login form.